### PR TITLE
Kanvi/Add C++ unit test for BatchMatMul Op

### PR DESCRIPTION
**Pull request by kanvi-nervana**

### CODE REVIEW

**Comment by kanvi-nervana** [comment]
TESTNOW

**Comment by kanvi-nervana** [comment]
TESTNOW

**Comment by kanvi-nervana** [comment]
TESTNOW

File: test/test_math_ops.cpp [hunk 1]
```diff
@@ -389,6 +389,113 @@ TEST(MathOps, SquaredDifferenceBroadcasting) {
...
+
+  vector<DataType> output_datatypes = {DT_FLOAT};
+
+  std::vector<Output> sess_run_fetchoutputs = {R};
+  OpExecuter opexecuter(root, "BatchMatMul", static_input_indexes,
+                        output_datatypes, sess_run_fetchoutputs);
+
+  opexecuter.RunTest();
+}
+
```
Comment (sayantan-nervana) [inline test/test_math_ops.cpp]: @kanvi-nervana Do we know which one of the 4 tests seg-faults?

**Comment by kanvi-nervana** [comment]
TESTNOW

**Pull request merged**